### PR TITLE
fixes on filter logs

### PR DESCRIPTION
**Pull request by cosmatudor**

### CODE REVIEW

File: blockchain/proxy.go [hunk 1]
```diff
@@ -736,6 +737,7 @@ func (ep *proxy) IsDataTrieMigrated(ctx context.Context, address sdkCore.Address
 // GetBlockBytesByNonce retrieves bytes of a block with a specific nonce
 func (ep *proxy) GetBlockBytesByNonce(ctx context.Context, shardID uint32, nonce uint64) ([]byte, error) {
```
Comment (iulianpascalau) [inline blockchain/proxy.go:738]: as suggested, please rename this function. Can also be made unexported

File: blockchain/proxy.go [hunk 2]
```diff
@@ -746,6 +748,7 @@ func (ep *proxy) GetBlockBytesByNonce(ctx context.Context, shardID uint32, nonce
 // GetBlockBytesByHash retrieves bytes of a block with a specific hash
 func (ep *proxy) GetBlockBytesByHash(ctx context.Context, shardID uint32, hash string) ([]byte, error) {
```
Comment (iulianpascalau) [inline blockchain/proxy.go:749]: also here

File: blockchain/proxy.go [hunk 3]
```diff
@@ -778,9 +786,58 @@ func (ep *proxy) FilterLogs(ctx context.Context, filter *sdkCore.FilterQuery) ([
...
+	filterAddresses := filter.Addresses
+	shardId := filter.ShardID
+
+	if len(filterAddresses) != 0 && shardId.HasValue {
+		shardIdFromAddresses, err := ep.computeShardIdFromAddresses(ctx, filterAddresses)
+		if err != nil {
+			return 0, err
+		}
+		if shardId.Value != shardIdFromAddresses {
+			return 0, errors.New("shardID from addresses does not match the provided shardID")
```
Comment (iulianpascalau) [inline blockchain/proxy.go:799]: might extract this and error on L816 in some variables

File: blockchain/proxy.go [hunk 4]
```diff
@@ -778,9 +786,58 @@ func (ep *proxy) FilterLogs(ctx context.Context, filter *sdkCore.FilterQuery) ([
...
+		}
+		return shardId.Value, nil
+	}
+
+	if len(filterAddresses) != 0 {
```
Comment (iulianpascalau) [inline blockchain/proxy.go:804]: this can be moved on L792 and L794-797 can be removed

File: blockchain/proxy.go [hunk 5]
```diff
@@ -778,9 +786,58 @@ func (ep *proxy) FilterLogs(ctx context.Context, filter *sdkCore.FilterQuery) ([
...
+		return 0, err
+	}
+
+	for _, address := range addresses {
+		addressShardId, err := ep.GetShardOfAddress(ctx, address)
+		if err != nil {
+			return 0, err
+		}
+		if addressShardId != shardId {
+			return 0, errors.New("addresses belong to different shards")
```
Comment (iulianpascalau) [inline blockchain/proxy.go:831]: can be a variable

File: blockchain/proxy_test.go [hunk 6]
```diff
@@ -1309,13 +1310,100 @@ func TestProxy_FilterLogs(t *testing.T) {
...
+			"https://test.org/block/by-nonce/21000005?withTxs=true&withLogs=true": blockResponseBytes,
+			"https://test.org/network/config":                                     httpNetworkConfig,
+		}
+
+		httpClient := createMockClientMultiResponse(responseMap)
+		args := createMockArgsProxy(httpClient)
+		ep, _ := NewProxy(args)
+
+		res, err := ep.FilterLogs(context.Background(), invalidFilter)
+		assert.Equal(t, errors.New("addresses belong to different shards"), err)
```
Comment (iulianpascalau) [inline blockchain/proxy_test.go:1367]: should use the new var for this test

File: blockchain/proxy_test.go [hunk 7]
```diff
@@ -1309,13 +1310,100 @@ func TestProxy_FilterLogs(t *testing.T) {
...
+			"https://test.org/block/by-nonce/21000005?withTxs=true&withLogs=true": blockResponseBytes,
+			"https://test.org/network/config":                                     httpNetworkConfig,
+		}
+
+		httpClient := createMockClientMultiResponse(responseMap)
+		args := createMockArgsProxy(httpClient)
+		ep, _ := NewProxy(args)
+
+		res, err := ep.FilterLogs(context.Background(), invalidFilter)
+		assert.Equal(t, errors.New("shardID from addresses does not match the provided shardID"), err)
```
Comment (iulianpascalau) [inline blockchain/proxy_test.go:1396]: should use the new var for this test

**Review by iulianpascalau — APPROVED** [review]

**Review by evelinemolnar — APPROVED** [review]

File: blockchain/testdata/network_config_data.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      "erd_num_nodes_in_shard": 400,
+      "erd_num_shards_without_meta": 3,
+      "erd_round_duration": 6000,
+      "erd_shard_consensus_group_size": 63,
+      "erd_start_time": 1596117600
+    }
+  },
+  "error": "",
+  "code": "successful"
+}
```
Comment (sstanculeanu) [inline blockchain/testdata/network_config_data.json:22]: missing new line

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**